include/Core/Posts/Metabox/DeletePostsByCategoryMetabox.php

```diff
@@ -0,0 +1,165 @@
+		$this->messages      = array(
+			'box_label' => __( 'By Post Category', 'bulk-delete' ),
+			'scheduled' => __( 'The selected posts are scheduled for deletion', 'bulk-delete' ),
+		);
+	}
+
+	/**
+	 * Render Delete posts by category box.
+	 */
+	public function render() {
+        ?>
+        <!-- Category Start-->
+        <h4><?php _e( 'Select the post type from which you want to delete posts by category', 'bulk-delete' ); ?></h4>
+        <fieldset class="options">
+            <table class="optiontable">
+				<?php bd_render_post_type_dropdown( 'cats' ); ?>
+            </table>
+
+            <h4><?php _e( 'Select the categories from which you wan to delete posts', 'bulk-delete' ); ?></h4>
+            <p><?php _e( 'Note: The post count below for each category is the total number of posts in that category, irrespective of post type', 'bulk-delete' ); ?>.</p>
+
+			<?php
+			/**
+			 * Filter to modify select2 ajax call category limit.
+			 *
+			 * @param int $cat_limit
+			 */
+			$bd_select2_ajax_limit_categories = apply_filters( 'bd_select2_ajax_limit_categories', $this->cat_limit );
+
+			$categories = get_categories( array(
+					'hide_empty'    => false,
+					'number'        => $bd_select2_ajax_limit_categories,
+				)
+			);
+			?>
+            <table class="form-table">
+                <tr>
+                    <td scope="row">
+						<?php if( count($categories) >= $bd_select2_ajax_limit_categories ){?>
+                            <select class="select2Ajax" name="smbd_cats[]" data-taxonomy="category" multiple data-placeholder="<?php _e( 'Select Categories', 'bulk-delete' ); ?>">
+                                <option value="all" selected="selected"><?php _e( 'All Categories', 'bulk-delete' ); ?></option>
+                            </select>
+						<?php }else{?>
+                            <select class="select2" name="smbd_cats[]" multiple data-placeholder="<?php _e( 'Select Categories', 'bulk-delete' ); ?>">
+                                <option value="all" selected="selected"><?php _e( 'All Categories', 'bulk-delete' ); ?></option>
+								<?php foreach ( $categories as $category ) { ?>
+                                    <option value="<?php echo $category->cat_ID; ?>"><?php echo $category->cat_name, ' (', $category->count, ' ', __( 'Posts', 'bulk-delete' ), ')'; ?></option>
+								<?php } ?>
+                            </select>
+						<?php }?>
+                    </td>
+                </tr>
+            </table>
+
+			<table class="optiontable">
+				<?php
+				$this->render_filtering_table_header();
+				$this->render_restrict_settings();
+				$this->render_delete_settings();
+				$this->render_private_post_settings();
+				$this->render_limit_settings();
+				$this->render_cron_settings();
+				?>
+			</table>
+
+		</fieldset>
+<?php
+		$this->render_submit_button( 'delete_posts_by_category' );
```

Comment:
The `render_submit_button` function doesn't take any parameters. Check its phpdoc